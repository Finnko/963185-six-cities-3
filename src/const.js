const Feature = {
  WIFI: `Wi-Fi`,
  WASHINGMACHINE: `Washing machine`,
  TOWELS: `Towels`,
  HEATING: `Heating`,
  COFFEEMACHINE: `Coffee machine`,
  BABYSET: `Baby seat`,
  KITCHEN: `Kitchen`,
  DISHWASHER: `Dishwasher`,
  CABELTV: `Cabel TV`,
  FRIDGE: `Fridge`
};

const features = [
  Feature.WIFI,
  Feature.WASHINGMACHINE,
  Feature.HEATING,
  Feature.COFFEEMACHINE,
  Feature.BABYSET,
  Feature.DISHWASHER,
  Feature.CABELTV,
  Feature.FRIDGE
];

const SortType = {
  POPULAR: `Popular`,
  PRICE_TO_HIGH: `Price: low to high`,
  PRICE_TO_LOW: `Price: high to low`,
  TOPRATED: `Top rated first`
};

const sortTypes = [
  SortType.POPULAR,
  SortType.PRICE_TO_HIGH,
  SortType.PRICE_TO_LOW,
  SortType.TOPRATED
];

const REVIEW_RATING_VALUES = [
  {value: 5, title: `perfect`},
  {value: 4, title: `good`},
  {value: 3, title: `not bad`},
  {value: 2, title: `badly`},
  {value: 1, title: `terribly`},
];

const AppViewMode = {
  MAIN: `Main`,
  DETAILS: `Details`
};

const INITIAL_CITY = {
  "name": `Amsterdam`,
  "coords": {
    "latitude": 52.37454,
    "longitude": 4.897976,
    "zoom": 13
  }
};

const AuthorizationStatus = {
  NO_AUTH: `NO_AUTH`,
  AUTH: `AUTH`,
};

const AppPaths = {
  getLogin: () => `/login`,
  getOffers: () => `/hotels`,
  getOffer: (id) => `/offer/${id}`,
  getRoot: () => `/`,
};


const ServerError = {
  UNAUTHORIZED: 401,
  BAD_REQUEST: 400,
};

const END_POINT = `https://htmlacademy-react-3.appspot.com/six-cities`;
const TIMEOUT = 1000 * 5;

export {features, SortType, sortTypes, AppPaths, AppViewMode, END_POINT,
  TIMEOUT, INITIAL_CITY, AuthorizationStatus, ServerError, REVIEW_RATING_VALUES
};
