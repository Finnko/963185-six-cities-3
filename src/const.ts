const END_POINT = 'https://6.react.pages.academy/six-cities';
const TIMEOUT = 1000 * 5;

const DEFAULT_CITIES = [
  'Paris',
  'Cologne',
  'Brussels',
  'Amsterdam',
  'Hamburg',
  'Dusseldorf',
];

const DEFAULT_CITY = {
  name: 'Paris',
  location: {
    latitude: 48.85661,
    longitude: 2.351499,
    zoom: 13,
  }
};

const Features: { [unit: string]: string } = {
  WIFI: 'Wi-Fi',
  WASHING_MACHINE: 'Washing machine',
  TOWELS: 'Towels',
  HEATING: 'Heating',
  COFFEE_MACHINE: 'Coffee machine',
  BABY_SEAT: 'Baby seat',
  KITCHEN: 'Kitchen',
  DISHWASHER: 'Dishwasher',
  CABLE_TV: 'Cabel TV',
  FRIDGE: 'Fridge',
};

const SortType: { [unit: string]: string } = {
  POPULAR: 'Popular',
  PRICE_TO_HIGH: 'Price: low to high',
  PRICE_TO_LOW: 'Price: high to low',
  TOP_RATED: 'Top rated first',
};

// const REVIEW_RATING_VALUES = [
//   {value: 5, title: 'perfect'},
//   {value: 4, title: 'good'},
//   {value: 3, title: 'not bad'},
//   {value: 2, title: 'badly'},
//   {value: 1, title: 'terribly'},
// ];

const AuthorizationStatus: { [unit: string]: string } = {
  NO_AUTH: 'NO_AUTH',
  AUTH: 'AUTH',
};

const ServerError: { [unit: string]: number} = {
  UNAUTHORIZED: 401,
  BAD_REQUEST: 400,
};


export {
  Features,
  SortType,
  END_POINT,
  TIMEOUT,
  DEFAULT_CITIES,
  DEFAULT_CITY,
  AuthorizationStatus,
  ServerError,
};
