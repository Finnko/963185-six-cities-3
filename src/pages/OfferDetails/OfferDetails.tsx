

import React, { useEffect } from 'react';
import {useDispatch, useSelector} from 'react-redux';
import { Redirect, useParams } from 'react-router-dom';

import { RootState } from '../../app/rootReducer';
import Header from '../../components/Header/Header';
import Offer from '../../components/Offer/Offer.tsx';
import OfferGallery from '../../components/OfferGallery/OfferGallery';
import { AppPaths } from '../../const';
import { fetchOffer } from '../../features/offers/offersSlice';
import { selectOffer } from '../../features/offers/selectors';
import Map from '../Map/map.jsx';
import OffersList from '../OffersList/offers-list.jsx';
import ReviewsList from '../ReviewsList/reviews-list.jsx';

interface OfferRouteParams {
  id: string;
}

const OfferDetails: React.FC = () => {
  const dispatch = useDispatch();
  const { id } = useParams<OfferRouteParams>();
  const offer = useSelector((state: RootState) => selectOffer(state, id));

  useEffect(() => {
    if (!offer) {
      dispatch(fetchOffer(id));
    }
  }, [dispatch, id, offer]);

  if (!offer) {
    return <Redirect to={AppPaths.ROOT}/>;
  }

  const { images } = offer;

  return (
    <div className="page">
      <Header />

      <main className="page__main page__main--property">
        <section className="property">
          <OfferGallery images={images}/>
          <div className="property__container container">
            <div className="property__wrapper">
              <Offer offer={offer}/>
              {/* <ReviewsList reviews={reviews}/>*/}
            </div>
          </div>
          {/* <Map isNearByView currentCity={currentCity} offers={nearByOffers}/>*/}
        </section>

        {/* <div className="container">*/}
        {/*  <section className="near-places places">*/}
        {/*    <h2 className="near-places__title">Other places in the neighbourhood</h2>*/}
        {/*    <OffersList offersCards={nearByOffers} isNearByView/>*/}
        {/*  </section>*/}
        {/* </div>*/}
      </main>
    </div>
  );
};

export default OfferDetails;
