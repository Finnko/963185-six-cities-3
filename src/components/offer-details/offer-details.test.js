import React from "react";
import renderer from "react-test-renderer";
import OfferDetails from "./offer-details.jsx";
import {FEATURES} from '../../const';
import configureStore from 'redux-mock-store';
import {Provider} from 'react-redux';
import {BrowserRouter} from 'react-router-dom';

const mockDate = new Date(1583591483969).valueOf();
const mock = {
  id: `id_11`,
  title: `Beautiful & luxurious apartment at great location`,
  price: `120`,
  type: `Apartment`,
  promoImage: `img/apartment-01.jpg`,
  isPremium: true,
  rating: 2.5,
  images: [
    `img/room.jpg`,
    `img/apartment-01.jpg`,
    `img/apartment-01.jpg`,
  ],
  roomsCount: 2,
  guestsCount: 4,
  features: [`Wi-Fi`, `Baby seat`, `Cabel TV`, `Fridge`],
  owner: {
    name: `Dominika`,
    avatar: `img/avatar-angelina.jpg`,
    description: [`A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.`],
    isPro: true
  },
  reviews: [
    {
      id: 1,
      name: `Max`,
      avatar: `/img/avatar-max.jpg`,
      rating: 1.5,
      description: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.`,
      date: mockDate
    },
    {
      id: 2,
      name: `Alexey`,
      avatar: `/img/avatar-max.jpg`,
      rating: 0.5,
      description: `The building is green and from 18th century.`,
      date: mockDate
    },
    {
      id: 3,
      name: `Anastasia`,
      avatar: `/img/avatar-max.jpg`,
      rating: 4.5,
      description: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.`,
      date: mockDate
    }]
};
const mocks = [
  {
    id: `id_11`,
    city: `Amsterdam`,
    title: `Beautiful & luxurious apartment at great location`,
    price: `120`,
    type: `Apartment`,
    promoImage: `/img/apartment-01.jpg`,
    isPremium: true,
    rating: 1.5,
    images: [
      `/img/room.jpg`,
      `/img/apartment-01.jpg`,
      `/img/apartment-02.jpg`,
      `/img/apartment-03.jpg`,
      `/img/studio-01.jpg`,
      `/img/apartment-01.jpg`,
    ],
    roomsCount: 2,
    guestsCount: 3,
    features: FEATURES,
    coords: [52.3909553943508, 4.85309666406198],
    owner: {
      name: `Angelina`,
      avatar: `/img/avatar-angelina.jpg`,
      description: [
        `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.`,
        `An independent House, strategically located between Rembrand Square and National Opera, but where the bustle of the city comes to rest in this alley flowery and colorful.`
      ],
      isPro: true
    },
    reviews: [
      {
        id: 1,
        name: `Max`,
        avatar: `/img/avatar-max.jpg`,
        rating: 1.5,
        description: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.`,
        date: mockDate
      },
      {
        id: 2,
        name: `Alexey`,
        avatar: `/img/avatar-max.jpg`,
        rating: 0.5,
        description: `The building is green and from 18th century.`,
        date: mockDate
      },
      {
        id: 3,
        name: `Anastasia`,
        avatar: `/img/avatar-max.jpg`,
        rating: 4.5,
        description: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.`,
        date: mockDate
      }]
  },
  {
    id: `id_12`,
    city: `Dusseldorf`,
    title: `Beautiful & luxurious apartment at great location`,
    price: `70`,
    type: `Apartment`,
    promoImage: `/img/apartment-01.jpg`,
    isPremium: false,
    rating: 3.5,
    images: [
      `/img/room.jpg`,
      `/img/apartment-01.jpg`,
      `/img/apartment-02.jpg`,
    ],
    roomsCount: 1,
    guestsCount: 2,
    features: FEATURES,
    coords: [52.3909553943508, 4.86309666406198],
    owner: {
      name: `Angelina 2`,
      avatar: `/img/avatar-angelina.jpg`,
      description: [
        `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.`,
        `An independent House, strategically located between Rembrand Square and National Opera, but where the bustle of the city comes to rest in this alley flowery and colorful.`
      ],
      isPro: true
    },
    reviews: [
      {
        id: 1,
        name: `Max`,
        avatar: `/img/avatar-max.jpg`,
        rating: 1.5,
        description: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.`,
        date: mockDate
      },
      {
        id: 3,
        name: `Anastasia`,
        avatar: `/img/avatar-max.jpg`,
        rating: 4.5,
        description: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.`,
        date: mockDate
      }]
  },
  {
    id: `id_13`,
    city: `Amsterdam`,
    title: `Beautiful & luxurious apartment at great location`,
    price: `125`,
    type: `Apartment`,
    promoImage: `/img/apartment-01.jpg`,
    isPremium: true,
    rating: 4.5,
    images: [
      `/img/room.jpg`,
      `/img/apartment-01.jpg`,
      `/img/apartment-02.jpg`,
      `/img/apartment-03.jpg`,
      `/img/studio-01.jpg`,
      `/img/apartment-01.jpg`,
    ],
    roomsCount: 3,
    guestsCount: 2,
    features: FEATURES,
    coords: [52.3909553943508, 4.84309666406198],
    owner: {
      name: `Angelina 3`,
      avatar: `/img/avatar-angelina.jpg`,
      description: [
        `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.`,
        `An independent House, strategically located between Rembrand Square and National Opera, but where the bustle of the city comes to rest in this alley flowery and colorful.`
      ],
      isPro: false
    },
    reviews: [
      {
        id: 1,
        name: `Max`,
        avatar: `/img/avatar-max.jpg`,
        rating: 1.5,
        description: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.`,
        date: mockDate
      }]
  },
];
const nearByOfferMocks = [
  {
    id: `id_12`,
    city: `Dusseldorf`,
    title: `Beautiful & luxurious apartment at great location`,
    price: `70`,
    type: `Apartment`,
    promoImage: `/img/apartment-01.jpg`,
    isPremium: false,
    rating: 3.5,
    images: [
      `/img/room.jpg`,
      `/img/apartment-01.jpg`,
      `/img/apartment-02.jpg`,
    ],
    roomsCount: 1,
    guestsCount: 2,
    features: FEATURES,
    coords: [52.3909553943508, 4.86309666406198],
    owner: {
      name: `Angelina 2`,
      avatar: `/img/avatar-angelina.jpg`,
      description: [
        `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.`,
        `An independent House, strategically located between Rembrand Square and National Opera, but where the bustle of the city comes to rest in this alley flowery and colorful.`
      ],
      isPro: true
    },
    reviews: [
      {
        id: 1,
        name: `Max`,
        avatar: `/img/avatar-max.jpg`,
        rating: 1.5,
        description: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.`,
        date: mockDate
      },
      {
        id: 3,
        name: `Anastasia`,
        avatar: `/img/avatar-max.jpg`,
        rating: 4.5,
        description: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.`,
        date: mockDate
      }]
  },
  {
    id: `id_13`,
    city: `Amsterdam`,
    title: `Beautiful & luxurious apartment at great location`,
    price: `125`,
    type: `Apartment`,
    promoImage: `/img/apartment-01.jpg`,
    isPremium: true,
    rating: 4.5,
    images: [
      `/img/room.jpg`,
      `/img/apartment-01.jpg`,
      `/img/apartment-02.jpg`,
      `/img/apartment-03.jpg`,
      `/img/studio-01.jpg`,
      `/img/apartment-01.jpg`,
    ],
    roomsCount: 3,
    guestsCount: 2,
    features: FEATURES,
    coords: [52.3909553943508, 4.84309666406198],
    owner: {
      name: `Angelina 3`,
      avatar: `/img/avatar-angelina.jpg`,
      description: [
        `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.`,
        `An independent House, strategically located between Rembrand Square and National Opera, but where the bustle of the city comes to rest in this alley flowery and colorful.`
      ],
      isPro: false
    },
    reviews: [
      {
        id: 1,
        name: `Max`,
        avatar: `/img/avatar-max.jpg`,
        rating: 1.5,
        description: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.`,
        date: mockDate
      }]
  },
];

const mockStore = configureStore([]);

const store = mockStore({
  offers: mocks,
  chosenCity: `Amsterdam`,
  cities: [`Amsterdam`, `Cologne`, `Brussels`, `Dusseldorf`],
  currentOffers: [mocks[0], mocks[2]]
});

it(`Should Property component render correctly`, () => {
  const tree = renderer.create(
      <Provider store={store}>
        <BrowserRouter>
          <OfferDetails offer={mock} nearByOffers={nearByOfferMocks}/>
        </BrowserRouter>
      </Provider>)
  .toJSON();

  expect(tree).toMatchSnapshot();
});
