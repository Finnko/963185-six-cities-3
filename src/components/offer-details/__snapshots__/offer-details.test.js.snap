// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should OfferDetails component render correctly 1`] = `
Array [
  <section
    className="property"
  >
    <div
      className="property__gallery-container container"
    >
      <div
        className="property__gallery"
      >
        <div
          className="property__image-wrapper"
        >
          <img
            alt="Photo studio"
            className="property__image"
            src="img/room.jpg"
          />
        </div>
        <div
          className="property__image-wrapper"
        >
          <img
            alt="Photo studio"
            className="property__image"
            src="img/apartment-01.jpg"
          />
        </div>
        <div
          className="property__image-wrapper"
        >
          <img
            alt="Photo studio"
            className="property__image"
            src="img/apartment-01.jpg"
          />
        </div>
      </div>
    </div>
    <div
      className="property__container container"
    >
      <div
        className="property__wrapper"
      >
        <div
          className="property__mark"
        >
          <span>
            Premium
          </span>
        </div>
        <div
          className="property__name-wrapper"
        >
          <h1
            className="property__name"
          >
            Beautiful & luxurious apartment at great location
          </h1>
          <button
            className="property__bookmark-button button"
            type="button"
          >
            <svg
              className="property__bookmark-icon"
              height="33"
              width="31"
            >
              <use
                xlinkHref="#icon-bookmark"
              />
            </svg>
            <span
              className="visually-hidden"
            >
              To bookmarks
            </span>
          </button>
        </div>
        <div
          className="property__rating rating"
        >
          <div
            className="property__stars rating__stars"
          >
            <span
              style={
                Object {
                  "width": "50%",
                }
              }
            />
            <span
              className="visually-hidden"
            >
              Rating
            </span>
          </div>
          <span
            className="property__rating-value rating__value"
          >
            2.5
          </span>
        </div>
        <ul
          className="property__features"
        >
          <li
            className="property__feature property__feature--entire"
          >
            Apartment
          </li>
          <li
            className="property__feature property__feature--bedrooms"
          >
            2
             Bedrooms
          </li>
          <li
            className="property__feature property__feature--adults"
          >
            Max 
            4
             adults
          </li>
        </ul>
        <div
          className="property__price"
        >
          <b
            className="property__price-value"
          >
            €
            120
          </b>
          <span
            className="property__price-text"
          >
             night
          </span>
        </div>
        <div
          className="property__inside"
        >
          <h2
            className="property__inside-title"
          >
            What's inside
          </h2>
          <ul
            className="property__inside-list"
          >
            <li
              className="property__inside-item"
            >
              Wi-Fi
            </li>
            <li
              className="property__inside-item"
            >
              Baby seat
            </li>
            <li
              className="property__inside-item"
            >
              Cabel TV
            </li>
            <li
              className="property__inside-item"
            >
              Fridge
            </li>
          </ul>
        </div>
        <div
          className="property__host"
        >
          <h2
            className="property__host-title"
          >
            Meet the host
          </h2>
          <div
            className="property__host-user user"
          >
            <div
              className="property__avatar-wrapper user__avatar-wrapper property__avatar-wrapper--pro "
            >
              <img
                alt="Host avatar"
                className="property__avatar user__avatar"
                height="74"
                src="img/avatar-angelina.jpg"
                width="74"
              />
            </div>
            <span
              className="property__user-name"
            >
              Dominika
            </span>
          </div>
          <div
            className="property__description"
          >
            <p
              className="property__text"
            >
              A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.
            </p>
          </div>
        </div>
        <section
          className="property__reviews reviews"
        >
          <h2
            className="reviews__title"
          >
            Reviews · 
            <span
              className="reviews__amount"
            >
              3
            </span>
          </h2>
          <ul
            className="reviews__list"
          >
            <ul
              className="reviews__list"
            >
              <li
                className="reviews__item"
              >
                <div
                  className="reviews__user user"
                >
                  <div
                    className="reviews__avatar-wrapper user__avatar-wrapper"
                  >
                    <img
                      alt="Reviews avatar"
                      className="reviews__avatar user__avatar"
                      height="54"
                      src="/img/avatar-max.jpg"
                      width="54"
                    />
                  </div>
                  <span
                    className="reviews__user-name"
                  >
                    Max
                  </span>
                </div>
                <div
                  className="reviews__info"
                >
                  <div
                    className="reviews__rating rating"
                  >
                    <div
                      className="reviews__stars rating__stars"
                    >
                      <span
                        style={
                          Object {
                            "width": 30,
                          }
                        }
                      />
                      <span
                        className="visually-hidden"
                      >
                        Rating
                      </span>
                    </div>
                  </div>
                  <p
                    className="reviews__text"
                  >
                    A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.
                  </p>
                  <time
                    className="reviews__time"
                    dateTime="2020-03-07"
                  >
                    March, 2020
                  </time>
                </div>
              </li>
            </ul>
            <ul
              className="reviews__list"
            >
              <li
                className="reviews__item"
              >
                <div
                  className="reviews__user user"
                >
                  <div
                    className="reviews__avatar-wrapper user__avatar-wrapper"
                  >
                    <img
                      alt="Reviews avatar"
                      className="reviews__avatar user__avatar"
                      height="54"
                      src="/img/avatar-max.jpg"
                      width="54"
                    />
                  </div>
                  <span
                    className="reviews__user-name"
                  >
                    Alexey
                  </span>
                </div>
                <div
                  className="reviews__info"
                >
                  <div
                    className="reviews__rating rating"
                  >
                    <div
                      className="reviews__stars rating__stars"
                    >
                      <span
                        style={
                          Object {
                            "width": 10,
                          }
                        }
                      />
                      <span
                        className="visually-hidden"
                      >
                        Rating
                      </span>
                    </div>
                  </div>
                  <p
                    className="reviews__text"
                  >
                    The building is green and from 18th century.
                  </p>
                  <time
                    className="reviews__time"
                    dateTime="2020-03-07"
                  >
                    March, 2020
                  </time>
                </div>
              </li>
            </ul>
            <ul
              className="reviews__list"
            >
              <li
                className="reviews__item"
              >
                <div
                  className="reviews__user user"
                >
                  <div
                    className="reviews__avatar-wrapper user__avatar-wrapper"
                  >
                    <img
                      alt="Reviews avatar"
                      className="reviews__avatar user__avatar"
                      height="54"
                      src="/img/avatar-max.jpg"
                      width="54"
                    />
                  </div>
                  <span
                    className="reviews__user-name"
                  >
                    Anastasia
                  </span>
                </div>
                <div
                  className="reviews__info"
                >
                  <div
                    className="reviews__rating rating"
                  >
                    <div
                      className="reviews__stars rating__stars"
                    >
                      <span
                        style={
                          Object {
                            "width": 90,
                          }
                        }
                      />
                      <span
                        className="visually-hidden"
                      >
                        Rating
                      </span>
                    </div>
                  </div>
                  <p
                    className="reviews__text"
                  >
                    A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.
                  </p>
                  <time
                    className="reviews__time"
                    dateTime="2020-03-07"
                  >
                    March, 2020
                  </time>
                </div>
              </li>
            </ul>
          </ul>
        </section>
      </div>
    </div>
    <section
      className="map property__map"
    >
      <div
        id="map"
        style={
          Object {
            "height": "100%",
          }
        }
      />
    </section>
  </section>,
  <div
    className="container"
  >
    <section
      className="near-places places"
    >
      <h2
        className="near-places__title"
      >
        Other places in the neighbourhood
      </h2>
      <div
        className="places__list near-places__list"
      >
        <article
          className="place-card near-places__card"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          
          <div
            className="place-card__image-wrapper near-places__image-wrapper"
          >
            <a
              href="#"
            >
              <img
                alt="Place image"
                className="place-card__image"
                height={200}
                src="/img/apartment-01.jpg"
                width={260}
              />
            </a>
          </div>
          <div
            className="place-card__info"
          >
            <div
              className="place-card__price-wrapper"
            >
              <div
                className="place-card__price"
              >
                <b
                  className="place-card__price-value"
                >
                  €
                  70
                </b>
                <span
                  className="place-card__price-text"
                >
                  / night
                </span>
              </div>
              <button
                className="place-card__bookmark-button button "
                type="button"
              >
                <svg
                  className="place-card__bookmark-icon"
                  height={19}
                  width={18}
                >
                  <use
                    xlinkHref="#icon-bookmark"
                  />
                </svg>
                <span
                  className="visually-hidden"
                >
                  To bookmarks
                </span>
              </button>
            </div>
            <div
              className="place-card__rating rating"
            >
              <div
                className="place-card__stars rating__stars"
              >
                <span
                  style={
                    Object {
                      "width": "70%",
                    }
                  }
                />
                <span
                  className="visually-hidden"
                >
                  Rating
                </span>
              </div>
            </div>
            <h2
              className="place-card__name"
            >
              <a
                href="/offer/id_12"
                onClick={[Function]}
              >
                Beautiful & luxurious apartment at great location
              </a>
            </h2>
            <p
              className="place-card__type"
            >
              Apartment
            </p>
          </div>
        </article>
        <article
          className="place-card near-places__card"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          <div
            className="place-card__mark"
          >
            <span>
              Premium
            </span>
          </div>
          <div
            className="place-card__image-wrapper near-places__image-wrapper"
          >
            <a
              href="#"
            >
              <img
                alt="Place image"
                className="place-card__image"
                height={200}
                src="/img/apartment-01.jpg"
                width={260}
              />
            </a>
          </div>
          <div
            className="place-card__info"
          >
            <div
              className="place-card__price-wrapper"
            >
              <div
                className="place-card__price"
              >
                <b
                  className="place-card__price-value"
                >
                  €
                  125
                </b>
                <span
                  className="place-card__price-text"
                >
                  / night
                </span>
              </div>
              <button
                className="place-card__bookmark-button button "
                type="button"
              >
                <svg
                  className="place-card__bookmark-icon"
                  height={19}
                  width={18}
                >
                  <use
                    xlinkHref="#icon-bookmark"
                  />
                </svg>
                <span
                  className="visually-hidden"
                >
                  To bookmarks
                </span>
              </button>
            </div>
            <div
              className="place-card__rating rating"
            >
              <div
                className="place-card__stars rating__stars"
              >
                <span
                  style={
                    Object {
                      "width": "90%",
                    }
                  }
                />
                <span
                  className="visually-hidden"
                >
                  Rating
                </span>
              </div>
            </div>
            <h2
              className="place-card__name"
            >
              <a
                href="/offer/id_13"
                onClick={[Function]}
              >
                Beautiful & luxurious apartment at great location
              </a>
            </h2>
            <p
              className="place-card__type"
            >
              Apartment
            </p>
          </div>
        </article>
      </div>
    </section>
  </div>,
]
`;
